### ISSUE TITLE
Creation event not fired for file after deleting (or moving it).

### ISSUE THREAD

**Issue by tolmasky** [atom/watcher#117]
### Description

1. Start observing a directory.
2. Create a file.
3. Move that file out of the directory outside of observation (on Mac this is detected, on Linux it is not)
4. Move file back into directory (on Mac this is not detected, on Linux it is, and then it "remembers" about the previous event).

I have a reproducible test case in RunKit here: https://runkit.com/tolmasky/moving-files-out-in-atom-watcher.

<img width="996" alt="screen shot 2018-02-11 at 7 22 44 pm" src="https://user-images.githubusercontent.com/23753/36082737-f7326d78-0f60-11e8-9b85-b7c49a32f5b4.png">

So to sum up, what I expect to see:

1. "file created/modified"
2. "file deleted"
3. "file created/modified"

What I see on Mac:
1. "file created/modified"
2. "file deleted"
3. nothing

What I see on Linux:
1. "file created/modified"
2. nothing
3. "file created/modified", "file deleted" <- in that order.
